src/history/history.rs

```diff
@@ -257,14 +259,19 @@ impl History {
             _ => "rank",
         };
 
+        let dir_filter = match &result_filter {
+            ResultFilter::Global => "%".to_string(),
+            ResultFilter::CurrentDirectory => dir.to_string(),
+        };
+
         let query: &str = &format!(
             "{} {} {} {}",
             "SELECT id, cmd, cmd_tpl, session_id, when_run, exit_code, selected, dir, rank,
                 age_factor, length_factor, exit_factor, recent_failure_factor,
                 selected_dir_factor, dir_factor, overlap_factor, immediate_overlap_factor,
                 selected_occurrences_factor, occurrences_factor, last_run
             FROM contextual_commands
-            WHERE cmd LIKE (:like)",
+            WHERE cmd LIKE (:like) AND dir LIKE (:dir)",
```

Comment:
When `result_filter` is `ResultFilter::Global`, can we not include the `AND dir LIKE (:dir)` SQL at all? This may be faster than `AND dir LIKE "%"`.

Reply:
I wasn't really sure of a clean way to do so without bringing in some conditionals around `named_params! { ":like": &like_query, ":limit": &num, ":dir": &dir_filter},` and that felt like it made things more difficult to understand.

Any ideas?

Reply:
I guess we should benchmark it. If you see no speed difference on a large database adding or omitting `AND dir LIKE (:dir)`, then this is fine. If there is one, we may want to add the conditional.

Reply:
I actually think the query planner will remove the redundant constraint. I’ll confirm on Monday

Reply:
The query planner does not remove our `WHERE dir LIKE '%"`, which is disappointing.

Benchmarks for 10k command history:

WITH LIKE '%": Run Time: real 0.083 user 0.014999 sys 0.014527
WITHOUT LIKE: Run Time: real 0.080 user 0.016174 sys 0.014996

At 10k entries, it's barely noticeable.

Tested with:

```shell
sqlite3 ~/Library/Application\ Support/McFly/history.db

.timer ON
 SELECT * FROM commands WHERE;
 SELECT * FROM commands WHERE dir LIKE '%';
```

Reply:
Out of curiosity, I wonder if `WHERE (:true_or_false OR dir LIKE '%')` would work and be faster, where `:true_or_false` is a flag when in directory mode.